cvat/apps/engine/cloud_provider.py

```diff
@@ -0,0 +1,301 @@
+            raise Exception('A resource {} unavailable'.format(self.name))
+
+    def is_object_exist(self, key_object):
+        waiter = self._client_s3.get_waiter('object_exists')
+        try:
+            waiter.wait(
+                Bucket=self._bucket,
+                Key=key_object,
+                WaiterConfig=self.waiter_config
+            )
+        except WaiterError:
+            raise Exception('A file {} unavailable'.format(key_object))
+
+    def upload_file(self, file_obj, file_name):
+        self._bucket.upload_fileobj(
+            Fileobj=file_obj,
+            Key=file_name,
+            Config=TransferConfig(max_io_queue=self.transfer_config['max_io_queue'])
+        )
+
+    def initialize_content(self):
+        files = self._bucket.objects.all()
+        self._files = [{
+            'name': item.key,
+        } for item in files]
+
+    def download_fileobj(self, key):
+        buf = BytesIO()
+        self.bucket.download_fileobj(
+            Key=key,
+            Fileobj=buf,
+            Config=TransferConfig(max_io_queue=self.transfer_config['max_io_queue'])
+        )
+        buf.seek(0)
+        return buf
+
+    def create(self):
+        try:
+            responce = self._bucket.create(
+                ACL='private',
+                CreateBucketConfiguration={
+                    'LocationConstraint': self.region,
+                },
+                ObjectLockEnabledForBucket=False
+            )
+            slogger.glob.info(
+                'Bucket {} has been created on {} region'.format(
+                    self.name,
+                    responce['Location']
+                ))
+        except Exception as ex:
+            msg = str(ex)
+            slogger.glob.info(msg)
+            raise Exception(msg)
+
+class AzureBlobContainer(_CloudStorage):
+    MAX_CONCURRENCY = 3
+    def __init__(self, container, account_name, sas_token=None):
+        super().__init__()
+        self._account_name = account_name
+        if sas_token:
+            self._blob_service_client = BlobServiceClient(account_url=self.account_url, credential=sas_token)
+        else:
+            self._blob_service_client = BlobServiceClient(account_url=self.account_url)
+        self._container_client = self._blob_service_client.get_container_client(container)
+
+    @property
+    def container(self):
+        return self._container_client
+
+    @property
+    def name(self):
+        return self._container_client.container_name
+
+    @property
+    def account_url(self):
+        return "{}.blob.core.windows.net".format(self._account_name)
+
+    def create(self):
+        try:
+            self._container_client.create_container(
+               metadata={
+                   'type' : 'created by CVAT',
+               },
+               public_access=PublicAccess.OFF
+            )
+        except ResourceExistsError:
+            msg = f"{self._container_client.container_name} alredy exists"
```

Comment:
@Marishka17 , There are multiple typos with `already` word. Probably you want to fix them all.